src/core/Track.cpp

```diff
@@ -697,8 +697,26 @@ void TrackContentObjectView::mousePressEvent( QMouseEvent * me )
 		m_tco->setJournalling( false );
 
 		setInitialMousePos( me->pos() );
-
-		if( me->x() < width() - RESIZE_GRIP_WIDTH )
+		SampleTCO * sTco = dynamic_cast<SampleTCO*>( m_tco );
+		if( me->x() < RESIZE_GRIP_WIDTH && sTco )
+		{
+			m_action = ResizeLeft;
+			m_oldTime = m_tco->startPosition();
+			QCursor c( Qt::SizeHorCursor );
+			QApplication::setOverrideCursor( c );
+			s_textFloat->setTitle( tr( "Current length" ) );
+			delete m_hint;
+			m_hint = TextFloat::displayMessage( tr( "Hint" ),
+					tr( "Press <%1> for free "
+							"resizing." ).arg(
+								#ifdef LMMS_BUILD_APPLE
+								"⌘"),
+								#else
+								"Ctrl"),
+								#endif
+					embed::getIconPixmap( "hint" ), 0 );
```

Comment:
I don't see how could this happen economical. Maybe you can give me a tip? All three cases are very similar. I don't know how to consolidate.